src/topology/basic.lean

```diff
@@ -1176,6 +1187,18 @@ lemma is_closed.preimage {f : α → β} (hf : continuous f) {s : set β} (h : i
   is_closed (f ⁻¹' s) :=
 continuous_iff_is_closed.mp hf s h
 
+lemma mem_closure_image {f : α → β} {x : α} {s : set α} (hf : continuous_at f x)
+  (hx : x ∈ closure s) : f x ∈ closure (f '' s) :=
+begin
+  rw [mem_closure_iff_nhds_ne_bot] at hx ⊢,
+  rw ← bot_lt_iff_ne_bot,
+  calc
+    ⊥   < map f (𝓝 x ⊓ principal s) : bot_lt_iff_ne_bot.mpr (map_neq_bot f hx)
```

Comment:
```suggestion
    ⊥   < map f (𝓝 x ⊓ principal s) : bot_lt_iff_ne_bot.mpr (ne_bot.map ⟨hx⟩ f).ne
```
is only slightly longer, and avoids the need for the `map_neq_bot` lemma. It looks like this is normally spelled as follows though, with `haveI : ne_bot _ := ⟨hx⟩` on a line above.
```suggestion
    ⊥   < map f (𝓝 x ⊓ principal s) : bot_lt_iff_ne_bot.mpr ne_bot.ne'
```

If there's a problem with `ne_bot` then I think we should design our way our of it, rather than ad ad-hoc lemmas like `map_neq_bot` to hack around it.